src/utils/helpers/should-component-update/should-component-update.js

```diff
@@ -35,7 +35,6 @@ function customCheck(current, next, key) {
   // allow isEqual to continue as normal
   if (Immutable.Iterable.isIterable(current)) {
     return Immutable.is(current, next);
-  } else {
-    return void 0;
   }
+  return undefined;
```

Comment:
I think return `void` was wrong in this case, `void ` should be used if a method is being called but nothing is being returned. 
`return null` indicates that no method is called and nothing is being returned either.